src/System.Collections.Specialized/src/System/Collections/Specialized/OrderedDictionary.cs

```diff
@@ -200,18 +203,25 @@ object ICollection.SyncRoot
         {
             get
             {
-                return objectsTable[key];
+                if (_objectsTable == null)
+                {
+                    return null;
+                }
+                EnsureObjectsTable();
```

Comment:
At this point you know _objectsTable is not null.

Reply:
Fixed.